packages/openapi-generator/src/parser/unique-naming.ts

```diff
@@ -2,107 +2,70 @@ import { UniqueNameGenerator, camelCase } from '@sap-cloud-sdk/util';
+ * @param names List of names to deduplicate.
+ * @param options Object containing options to configure the transformation.
+ * @param options.format Function to format the name. Defaults to camel case.
+ * @param options.reservedWords Reserved words that should be handled as duplicates.
+ * @returns Unique names in the given order.
  */
-export function ensureUniqueNames<ItemT, UniqueItemT>(
-  items: ItemT[],
-  nameHandler: {
-    getName?: (item: ItemT) => string;
-    transformItem?: (item: ItemT, name: string) => UniqueItemT;
-    formatName?: (name: string) => string;
+export function ensureUniqueNames(
+  names: string[],
+  options: {
+    format?: (name: string) => string;
     reservedWords?: string[];
   } = {}
-): UniqueItemT[] {
-  const {
-    getName = getNameDefault,
-    transformItem = transformItemDefault,
-    formatName = camelCase,
-    reservedWords = []
-  } = nameHandler;
+): string[] {
+  const { format: format = camelCase, reservedWords = [] } = options;
 
-  const uniqueItems = getCorrectlyNamedItems(
-    items,
-    getName,
-    formatName,
-    reservedWords
-  );
+  const correctNames = getCorrectNames(names, format, reservedWords);
 
   const nameGenerator = new UniqueNameGenerator('', [
     ...reservedWords,
-    ...uniqueItems.map(item => getName(item))
+    ...correctNames
   ]);
 
-  const uniqueNames = uniqueItems.map(item => getName(item));
-  return items.map(item => {
-    const name = getName(item);
-    if (uniqueNames.length && uniqueNames[0] === name) {
-      uniqueNames.shift();
-      return transformItem(item, name);
+  return names.map(name => {
+    if (correctNames.length && correctNames[0] === name) {
+      correctNames.shift();
+      return name;
     }
-    return transformItem(
-      item,
-      nameGenerator.generateAndSaveUniqueName(formatName(name))
-    );
-  });
+    return nameGenerator.generateAndSaveUniqueName(format(name));
+  }, {});
 }
 
 /**
- * Get the items within a list of items that won't have to be renamed.
- * Those are the items that have a unique name and a name that does not have to be formatted.
- * @param items Named items.
- * @param getName Function to get the name of an item.
- * @param formatName Function to transform the name when finding a unique name.
+ * Get the names within a list of names that won't have to be renamed.
+ * Those are the names that are unique and have a name that does not need to be formatted.
+ * @param names Names to check.
+ * @param format Function to format the name.
  * @param reservedWords Reserved words that should be handled as duplicates.
- * @returns An object containing the unique operations, denoted by `unique` and operations with (potentially) duplicate names, denoted by `duplicate`.
+ * @returns A list of names that do not need to be renamed.
  */
-function getCorrectlyNamedItems<ItemT>(
-  items: ItemT[],
-  getName: (item: ItemT) => string,
-  formatName: (name: string) => string,
+function getCorrectNames(
```

Comment:
like this helper methods with the other helper methods inside. Also wonder about the name but have no good alternative. `getNonConflictingNames()`

Reply:
but not sure

Reply:
I know that this method and the other `isReserve,isDuplicate,isFormatted` are only used internaly but are great candidates for nice small unit tests. Should we make them visible for tests?

Reply:
Hmm, I only moved isReserved, isDuplicate, isFormatted out, so that they are not executed if one of the others already returns true (isDuplicate and isFormatted go over "all" names every time they are called).
I think those functions are trivial and because ensureUniqueNames is tested it shouldn't be necessary to test them in my opinion.

Reply:
> like this helper methods with the other helper methods inside. Also wonder about the name but have no good alternative. `getNonConflictingNames()`

I don't like the negation in the name, but I think it is more on point than "correct". Will use it.